### ISSUE TITLE
Transfer Learning Mimarisi (ResNet50)

### ISSUE THREAD

**Issue by MesuttSahin** [MesuttSahin/teknofest_xray_sim#28]
###  BAĞIMLILIK DURUMU
* ** Girdi:** **Veri Mühendisi** ve **Veri Analisti**.
    * Analistin oluşturduğu `plots.py` (Grafik çizimi için).
    * Mühendisin hazırladığı `train.py` (Eğitim taslağı) ve `processed/train_list.csv` dosyaları.
* ** Çıktı:** Eğitilmiş Model (`.pth` dosyası) ve Loss Grafiği.

###  Görev Özeti
Önce **ResNet50** modelini projemize (14 Sınıf) uyarlayacak `model.py` dosyasını yazacaksın. Ardından Mühendisin yazdığı kodu çekip **EĞİTİMİ BAŞLATACAKSIN.** 

###  TEKNİK GÖREVLER

**A. Model Tasarımı (`src/models/model.py`):**
1.  **Modeli İndir:**
    * `torchvision.models.resnet50(weights='IMAGENET1K_V1')` kullan.
2.  **Katman Değişimi **
    * Modelin son katmanını (`fc`) sil.
    * Yerine `nn.Linear(2048, config.NUM_CLASSES)` ekle. (Sınıf sayısı 14 kalacak).
3.  **Forward:**
    * Standart akışı sağla.

**B. Eğitim**
1. İlk olarak dosya yapısını ayarla drive da data klasörünün altındaki resimleri data/raw a ekle.  
2.  **Pull:** Mühendisin kodlarını çek (`git pull`).
3.  **Review:** `src/train.py` dosyasını aç.
    * Senin yazdığın `model` sınıfını doğru import etmiş mi?
    * `Dataset` yolları doğru mu?
4.  **Run (Başlat):** Terminali aç ve komutu ver:
    ```bash
    python src/train.py
    ```
5.  **Takip:** Eğitim bitene kadar terminali izle. Hata çıkarsa `train.py` dosyasını düzeltme yetkisi sendedir.

###  Checklist
- [x] `config.NUM_CLASSES` (14) kullanıldı mı?
- [x] Model GPU (`config.DEVICE`) uyumlu mu?
- [x] `train.py` çalıştırıldı ve eğitim başladı mı?
- [x] Loss değerleri düşüyor mu? (Örn: 0.7 -> 0.5)
